src/ripple/app/main/LoadManager.cpp

```diff
@@ -129,23 +129,32 @@ void LoadManager::run ()
             auto const reportingIntervalSeconds = 10s;
             if (armed && (timeSpentDeadlocked >= reportingIntervalSeconds))
             {
+                constexpr auto deadlockTimeLimit = 90s;
+
                 // Report the deadlocked condition every 10 seconds
                 if ((timeSpentDeadlocked % reportingIntervalSeconds) == 0s)
                 {
-                    JLOG(journal_.warn())
-                        << "Server stalled for "
-                        << timeSpentDeadlocked.count() << " seconds.";
+                    if (timeSpentDeadlocked < deadlockTimeLimit)
+                    {
+                        JLOG(journal_.warn())
+                            << "Server stalled for "
+                            << timeSpentDeadlocked.count() << " seconds.";
+                    }
+                    else
+                    {
+                        JLOG(journal_.fatal())
+                            << "Deadlock detected. Deadlocked time: "
+                            << timeSpentDeadlocked.count()
+                            << "s JobQueue Overloaded: "
+                            << app_.getJobQueue().isOverloaded();
```

Comment:
Unlikely but I wanted to note that if changes ever make `deadlockTimeLimit` not divisible by `reportingIntervalSeconds`, we could potentially miss a fatal log. Say for instance that `reportingIntervalSeconds` is 12, and a beat is heard at 95 seconds.

Reply:
I was thinking the same thing about the modulus.  But, yes, I also think it's unlikely.  This particular thread wakes itself up once every second using the `alertable_sleep_until` on line 189.

Reply:
It's a good thought. I suppose we could keep track of last reported time and check if some duration has passed. But let's not do that in this PR; I'd like to keep this one relatively minimal.